bundles/de.uka.ipd.sdq.dialogs/src/de/uka/ipd/sdq/dialogs/selection/SelectEObjectDialog.java

```diff
@@ -85,11 +89,36 @@ public SelectEObjectDialog(final Shell parentShell, final String resourceName, f
         this.contentProvider = contentProvider;
 
         viewer.setContentProvider(contentProvider);
+        viewer.setComparator(createViewerComparator());
         viewer.setLabelProvider(labelProvider);
         viewer.setInput(input);
         viewer.expandAll();
     }
 
+    private static ViewerComparator createViewerComparator() {
+        return new ViewerComparator() {
+
+            private boolean isRootObject(Object o) {
+                Optional<EObject> eobject = Optional.ofNullable(o).filter(EObject.class::isInstance)
+                        .map(EObject.class::cast);
+                if (eobject.isPresent()) {
+                    return eobject.get().eContainer() == null;
+                }
+                return false;

```

Comment:
you could boil this down to a .filter() and subsequent .isPresent()

Reply:
I do not think that this is possible using a filter step without the assumption that `o` is an EObject:
* `eobject.filter(e -> e.eContainer() == null).isPresent()` will yield true even if `o` is no EObject, which means it cannot be an root element. Therefore, it is not correct.
* `!eobject.filter(e -> e.eContainer != null).isPresent()` also yields a wrong result if `o` is no EObject.

However, I think a map statement would yield the correct result. I did not come to this solution while writing the code but I will integrate it now:
`eobject.map(e -> e.eContainer == null).orElse(false)`

Reply:
`eobject.filter(e -> e.eContainer() == null).isPresent()` will yield true, iff `eobject.isPresent()` yields true (therefore o is an EObject) and `((EObject)o).eContainer==null`. Or am I mistaken?

Reply:
My statements made sense when I thought about it but you are right ;)